keps/1-kubernetes-enhancement-proposal-process.md

```diff
@@ -221,106 +218,34 @@ Metadata items:
-it is possible that a KEP would be retired if its functionality no longer provides
-sufficient value to the community.
+A KEP is has the following states
+
+- `Draft`: The KEP has been proposed and is actively being defined.
+- `Deferred`: The KEP is proposed but not actively being worked on.
+- `Approved`: Using lazy consensus, the approvers have approved this KEP.
+  It will move forward to implementation.
+- `Rejected`: The approvers and authors have decided that this KEP is not moving forward.
+  The KEP is kept around as a historical document.
+- `Withdrawn`: The KEP has been withdrawn by the authors.
+- `Final`: The KEP has been implemented and is no longer actively changed.
+- `Replaced`: The KEP has been replaced by a new KEP.
+  The `superseded-by` metadata value should point to the new KEP.
 
 ### Git and GitHub Implementation
 
-Practically an KEP would be implemented as a pull request to a central repository
-with the following example structure
-
-```
-├── 0000-kep-template.md
-├── CODEOWNERS
-├── index.md
-├── sig-architecture
-│   ├── deferred
-│   ├── orphaned
-│   └── retired
-├── sig-network
-│   ├── deferred
-│   ├── kube-dns
-│   ├── orphaned
-│   └── retired
-├── sig-node
-│   ├── deferred
-│   ├── kubelet
-│   ├── orphaned
-│   └── retired
-├── sig-release
-│   ├── deferred
-│   ├── orphaned
-│   └── retired
-├── sig-storage
-│   ├── deferred
-│   ├── orphaned
-│   └── retired
-├── unsorted-to-be-used-by-newcomers-only
-└── wg-resource-management
-    ├── deferred
-    ├── orphaned
-    └── retired
-```
-
-where each SIG or working group is given a top level directory with subprojects
-maintained by the SIG listed in sub directories. For newcomers to the community
-an `unsorted-to-be-used-by-newcomers-only` directory may be used before an KEP
-can be properly routed to a SIG although hopefully if discussion for a potential
-KEP begins on the mailing lists proper routing information will be provided to
-the KEP author. Additionally a top level index of KEPs may be helpful for people
-looking for a complete list of KEPs. There should be basic CI to ensure that an
-`index.md` remains up to date.
-
-Ideally no work would begin within the repositories of the Kubernetes organization
-before a KEP has been approved by the responsible SIG or working group. While the
-details of how SIGs organize their work is beyond the scope of this proposal one
-possibility would be for each charter SIG to create a top level repository within
-the Kubernetes org where implementation issues managed by that SIG would be filed.
+KEPs are checked into the community repo under the `/kep` directory.
+KEPs can either be put at the top level of that directory or in a SIG specific subdirectory.
+KEPs in SIG specific subdirectories have limited impact outside of the SIG and can leverage SIG specific OWNERS files.
+
+New KEPs can be checked in with a file name in the form of `draft-YYYYMMDD-my-title.md`.
```

Comment:
if one purpose of quickly checking in KEPs was to assign them a number, why have the intermediate filename? Also creation information is derivable from source control

Reply:
While source control is great for history, I'm not sure I want to lean on that completely.  I want to provide a way for users to easily visualize this and the tooling for that (static site generators) generally don't make that data available.  Also, if we ever move repos that data might be lost.  I'm looking at things like the RFC archive or the way that the W3C manages documents.  They don't rely on outside data sources for metadata.

Reply:
Wrt assigning a number.  I'm okay with having it assigned as part of the first PR.  I do worry that we may decide that some KEPs are a bad idea soon after creation and the numbers will be sparse.  But we can solve that problem when we get to it.

Reply:
filename information _should never_ be lost when moving repositories, nor should the history in my opinion although it may become a bit mangled. Both of the examples you mentioned for prior art on document management are from a time before distributed VCS systems were in wide use. I think we can cross the bridge of needing to store all the information within the file itself when and if we come to it. I think the sparseness is fine and that the only hard requirement was that the numbers be monotonically increasing

Reply:
I'm cool with assigning KEP numbers early.  Wrt relying on VCS metadata -- I'd rather not. It creates a more complicated tool chain and makes it harder to process the archive of KEPs without making assumptions about the VCS. It just feels like a layer violation to me.

Reply:
I don't think I agree that it makes the took chain more complicated. At some point I imagine you'll want to have a CI build for deploying say a website. Git is already part of our tool chain so I'm not really seeing the additional overhead. I don't imagine we'll _ever_ move away from git as the VCS despite what hosting provider we might use (self, GitHub, GitLab, Bitbucket) as it would be an incredibly disruptive change.